subsys/net/ip/net_context.c

```diff
@@ -2096,6 +2096,11 @@ static int sendto(struct net_pkt *pkt,
 		return -EBADF;
 	}
 
+	ret = bind_default(context);
```

Comment:
I would definitely suggest to call this only for UDP case, there's no need to do an extra call for TCP case. There're 2 ways to do that:

1. If we care about source addr to be bound for offloading case, then just add bind_default() block wrapped in `#if defined(CONFIG_NET_UDP)` after existing `#if defined(CONFIG_NET_TCP)` right below.
2. If we can cheat on offloading case, can reuse existing `if (net_context_get_ip_proto(context) == IPPROTO_UDP)` far below.

@jukkar : What do you think?

Reply:
Sure, makes sense.

Reply:
@jukkar: Thanks for the reply, but it's "either/or" type of question (either way 1 above, or way 2) ;-). Please suggest which way to go.

Reply:
@GAnthony : Maybe you can suggest here something too.

Reply:
I prefer the second one.